### ISSUE TITLE
Failed to download VS Code Server when Reopening folder (OFFLINE) to develop in a container

### ISSUE THREAD

**Issue by mbenezra** [microsoft/vscode-remote-release#2137]
I am attempting to use Remote Containers in an offline environment.

When you open a folder in VS Code which contains ./devcontainer/devcontainer.json, VS Code prompts you to reopen folder in a container - "Folder contains a dev container configuration file. Reopen folder to develop in a container (learn more)." 
 
When you select to reopen the folder in a container, VS Code starts building the image from the .devcontainer/Dockerfile specified in the ./devcontainer/devcontainer.json before starting the container from that image.

Since the .devcontainer/Dockerfile involves, apt-get and pip installs, this can not be done offline.

Instead I've built the image from .devcontainer/Dockerfile in an online environment and then saved the image, copied to the offline environment and loaded it as a docker image again. I then indicated in ./devcontainer/devcontainer.json that VS Code should use the "image" instead of the "Dockerfile" meaning it doesn't need to build the image offline since I've already done that online.

To do this I built an image called vscode-remote-air-gapped-anaconda in an online environment and moved it to the offline environment. My decontainer.json then points at the vscode-remote-air-gapped-anaconda instead of using the Dockerfile.

However when I try to open the folder in a container in the offline environment VS Code fails on "Failed to download VS Code Server" (see at the end of start up script below):

I should also note that the offline environment is Windows and the online environment is macOS.

[1 ms] Start: Resolving remote
[4 ms] Setting up container for folder or workspace: d:\path\to\folder
 
[30 ms] Start: Check Docker is running
[70 ms] Start: Run: git rev-parse --show-toplevel
[103 ms] Start: Starting container
[103 ms] Start: Run: docker run -a STDOUT -a STDERR --mount type=bind,source=d:\path\to\folder,target=/workspaces/folder,consistency=cached -l vsch.quality=stable -l vsch.remote.devPort=0 -l vsch.local.folder=d:\path\to\folder --entrypoint /bin/sh vscode-remote-air-gapped-anaconda -c echo Container started ;  while sleep 1; do :; done
[224 ms]
[1094 ms] Container started
 
[1101 ms] Start: Inspecting container
[1122 ms] Start: Run in container: uname -m
[1285 ms] x86_64
[1285 ms]
[1285 ms] Start: Run in container: (cat /etc/os-release || cat /usr/lib/os-release) 2>/dev/null
[1291 ms] PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
[1291 ms]
[1291 ms] Start: Run in container: cat /etc/passwd
[1294 ms] Start: Updating configuration state
[1295 ms] Start: Setup shutdown monitor
[1298 ms] Forking shutdown monitor: c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.94.0\dist\shutdownMonitorProcess \\.\pipe\vscode-remote-containers-fc55c3182d68dbae5834faa5e445e4b4cd3e7ea2-sock singleContainer Info
[1355 ms] Start: Run in container: test -d /root/.vscode-server
[1358 ms]
[1358 ms]
[1358 ms] Exit code 1
[1358 ms] Start: Run in container: test -d /root/.vscode-remote
[1361 ms]
[1361 ms]
[1361 ms] Exit code 1
[1361 ms] Start: Run in container: set -o noclobber ; mkdir -p '/root/.vscode-server/data/Machine' && { > '/root/.vscode-server/data/Machine/.writeMachineSettingsMarker' ; } 2> /dev/null
[1371 ms]
[1371 ms]
[1371 ms] Start: Run in container: mkdir -p '/root/.vscode-server/data/Machine' && cat >'/root/.vscode-server/data/Machine/settings.json' <<'settingsJSON'
[1376 ms]
[1376 ms]
[1378 ms] Start: Run in container: test -d /root/.vscode-server/bin/26076a4de974ead31f97692a0d32f90d735645c0
[1380 ms]
[1380 ms]
[1380 ms] Exit code 1
[1381 ms] Installing VS Code Server for commit 26076a4de974ead31f97692a0d32f90d735645c0
[1381 ms] Start: Run in container: mkdir -p /root/.vscode-server/bin/26076a4de974ead31f97692a0d32f90d735645c0_1578487035531
[1386 ms]
[1386 ms]
[1386 ms] Start: Downloading VS Code Server
[1508 ms] Failed to download VS Code Server (https://update.code.visualstudio.com/commit:26076a4de974ead31f97692a0d32f90d735645c0/server-linux-x64/stable): HTTP 403 - Forbidden

This is my ./devcontainer/devcontainer.json:

{
	"name": "Python 3 - Anaconda",
	"context": "..",
	"image": "vscode-remote-air-gapped-anaconda",

	"settings": { 
		"terminal.integrated.shell.linux": "/bin/bash",
		"python.pythonPath": "/opt/conda/bin/python",
		"python.linting.enabled": true,
		"python.linting.pylintEnabled": true,
		"python.linting.pylintPath": "/opt/conda/bin/pylint"
	},
}

This is my .devcontainer/Dockerfile:

FROM continuumio/anaconda3

ENV DEBIAN_FRONTEND=noninteractive

ARG USERNAME=vscode
ARG USER_UID=1000
ARG USER_GID=$USER_UID

COPY environment.yml* .devcontainer/noop.txt /tmp/conda-tmp/

RUN apt-get update \
    && apt-get -y install --no-install-recommends apt-utils dialog 2>&1 \
    && apt-get -y install git iproute2 procps iproute2 lsb-release \
    && /opt/conda/bin/pip install pylint \
    && if [ -f "/tmp/conda-tmp/environment.yml" ]; then /opt/conda/bin/conda env update -n base -f /tmp/conda-tmp/environment.yml; fi \
    && rm -rf /tmp/conda-tmp \
    && groupadd --gid $USER_GID $USERNAME \
    && useradd -s /bin/bash --uid $USER_UID --gid $USER_GID -m $USERNAME \
    && apt-get install -y sudo \
    && echo $USERNAME ALL=\(root\) NOPASSWD:ALL > /etc/sudoers.d/$USERNAME\
    && chmod 0440 /etc/sudoers.d/$USERNAME \
    && apt-get autoremove -y \
    && apt-get clean -y \
    && rm -rf /var/lib/apt/lists/*

ENV DEBIAN_FRONTEND=dialog

**Comment by egamma** [comment]
>However when I try to open the folder in a container in the offline environment VS Code fails on "Failed to download VS Code Server" (see at the end of start up script below)

This is expected, the matching VS Code Server for the locally installed VS Code is injected into the docker container, it is not built into the image.

Improving the off-line support is a duplicate of #1242
